src/mlpack/tests/main_tests/linear_regression_fit_test.cpp

```diff
@@ -0,0 +1,120 @@
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "main_test_fixture.hpp"
+
+#include "../test_catch_tools.hpp"
+#include "../catch.hpp"
+
+using namespace mlpack;
+
+BINDING_TEST_FIXTURE(LRFitTestFixture);
+
+/**
+ * Training a model with different regularization parameter and ensuring that
+ * predictions are different.
+ */
+TEST_CASE_METHOD(LRFitTestFixture, "LRFitDifferentLambdas",
+                 "[LinearRegressionFitMainTest][BindingTests]")
+{
+  // A required minimal difference between solutions.
+  const double delta = 0.1;
+
+  arma::mat trainX({1.0, 2.0, 3.0});
+  arma::mat testX({4.0});
+  arma::rowvec trainY({1.0, 4.0, 9.0});
+
+  SetInputParam("training", trainX);
+  SetInputParam("training_responses", trainY);
+  SetInputParam("lambda", 0.1);
+
+  // The first solution.
+  RUN_BINDING();
+  arma::rowvec preds1;
+  params.Get<LinearRegression*>("output_model")->Predict(testX,
+      preds1);
+  const double testY1 = preds1(0);
+
+  ResetSettings();
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("training_responses", std::move(trainY));
+  SetInputParam("lambda", 1.0);
+
+  // The second solution.
+  RUN_BINDING();
+  arma::rowvec preds2;
+  params.Get<LinearRegression*>("output_model")->Predict(testX,
+      preds2);
+  const double testY2 = preds2(0);
+
+  // Second solution has stronger regularization,
+  // so the predicted value should be smaller.
+  REQUIRE(testY1 - delta > testY2);
+}
+
+/**
+ * Ensuring that response size is checked.
+ */
+TEST_CASE_METHOD(LRFitTestFixture, "LRFitWrongResponseSizeTest",
+                 "[LinearRegressionFitMainTest][BindingTests]")
+{
+  constexpr int N = 10;
+  constexpr int D = 2;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  arma::rowvec trainY = arma::randu<arma::rowvec>(N + 3); // Wrong size.
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("training_responses", std::move(trainY));
+
+  Log::Fatal.ignoreInput = true;
+  REQUIRE_THROWS_AS(RUN_BINDING(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensuring that absence of responses is checked.
+ */
+TEST_CASE_METHOD(LRFitTestFixture, "LRFitNoResponses",
+                 "[LinearRegressionFitMainTest][BindingTests]")
+{
+  constexpr int N = 10;
+  constexpr int D = 1;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  SetInputParam("training", std::move(trainX));
+
+  Log::Fatal.ignoreInput = true;
+  REQUIRE_THROWS_AS(RUN_BINDING(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensuring that absence of training data is checked.
+ */
+TEST_CASE_METHOD(LRFitTestFixture, "LRFitNoTrainingData",
+                 "[LinearRegressionFitMainTest][BindingTests]")
+{
+  constexpr int N = 10;
+
+  arma::rowvec trainY = arma::randu<arma::rowvec>(N);
+  SetInputParam("training_responses", std::move(trainY));
+
+  Log::Fatal.ignoreInput = true;
+  REQUIRE_THROWS_AS(RUN_BINDING(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
```

Comment:
It could also be useful to check that an exception is thrown when negative regularization is passed.  I can't quickly think of anything else to add though.  Perhaps it's worth checking that when passing a 2-d matrix and no responses, everything succeeds?  (That is, the responses are taken to be the second row of data.)